lib/full-site-editing/templates.php

```diff
@@ -192,3 +192,94 @@ function set_unique_slug_on_create_template( $post_id ) {
 	}
 }
 add_action( 'save_post_wp_template', 'set_unique_slug_on_create_template' );
+
+/**
+ * Print the skip-link script & styles.
+ *
+ * @return void
+ */
+function gutenberg_the_skip_link() {
+
+	// Early exit if not an FSE theme.
+	if ( ! gutenberg_supports_block_templates() ) {
+		return;
+	}
+	?>
+
+	<?php
+	/**
+	 * Print the skip-link styles.
+	 */
+	?>
+	<style id="skip-link-styles">
+		.skip-link.screen-reader-text {
+			border: 0;
+			clip: rect(1px,1px,1px,1px);
+			clip-path: inset(50%);
+			height: 1px;
+			margin: -1px;
+			overflow: hidden;
+			padding: 0;
+			position: absolute !important;
+			width: 1px;
+			word-wrap: normal !important;
+		}
+
+		.skip-link.screen-reader-text:focus {
+			background-color: #eee;
+			clip: auto !important;
+			clip-path: none;
+			color: #444;
+			display: block;
+			font-size: 1em;
+			height: auto;
+			left: 5px;
+			line-height: normal;
+			padding: 15px 23px 14px;
+			text-decoration: none;
+			top: 5px;
+			width: auto;
+			z-index: 100000;
+		}
+	</style>
+	<?php
+	/**
+	 * Print the skip-link script.
+	 */
+	?>
+	<script>
+	( function() {
+		var skipLinkTarget = document.querySelector( 'main, .wp-block-post-title, .wp-block-query-loop, .wp-block-post-content, .entry-content' ),
```

Comment:
@aristath: I'm wondering why this list was expanded today. I was thinking that it would be easier to start conservatively with `main` and see — as FSE progresses — what other compelling selectors we'd like to add.

The rationale is this: starting small means that only users adhering to the `<main>`-based "contract" will benefit from this feature automatically. Conversely, casting a wide net by supporting multiple selectors from the beginning means that, if we want to iterate on the criteria for what constitutes the main content area, then we are either stuck supporting legacy selectors, or we remove them and break the trust of users who had come to expect the skip links feature. For example, as previously discussed, if we want to switch from DOM-based rules (querying for selectors) to block-based rules (detecting certain semantic blocks), switching from `querySelector( 'main' )` to a hypothetical `queryBlock( { name: 'core/group', tag: 'main' } )` would be fairly easy.

Anyway, that's my cautious advice, but I trust both you and @carolinan (who has just approved this PR) to make this call. :)

Reply:
Reverted the addition of extra elements :+1: 
expanding it in the future will be easy if and when we decide to do something like that :+1:

Reply:
Thanks for accommodating!